api/v1/atlasintegration_types.go

```diff
@@ -0,0 +1,187 @@
+// +kubebuilder:validation:XValidation:rule="!has(self.pagerDuty) || (self.type == 'PAGER_DUTY' && has(self.pagerDuty))",message="only PAGER_DUTY type may set pagerDuty fields"
+// +kubebuilder:validation:XValidation:rule="!has(self.slack) || (self.type == 'SLACK' && has(self.slack))",message="only SLACK type may set slack fields"
+// +kubebuilder:validation:XValidation:rule="!has(self.victorOps) || (self.type == 'VICTOR_OPS' && has(self.victorOps))",message="only VICTOR_OPS type may set victorOps fields"
+// +kubebuilder:validation:XValidation:rule="!has(self.webhook) || (self.type == 'WEBHOOK' && has(self.webhook))",message="only WEBHOOK type may set webhook fields"
+type AtlasIntegrationSpec struct {
+	ProjectDualReference `json:",inline"`
+
+	// ID of the integration in Atlas. May be omitted to create a new one.
+	// +kubebuilder:validation:Optional
+	ID *string `json:"id"`
+
+	// Type of the integration
+	// +kubebuilder:validation:Enum:=DATADOG;MICROSOFT_TEAMS;NEW_RELIC;OPS_GENIE;PAGER_DUTY;PROMETHEUS;SLACK;VICTOR_OPS;WEBHOOK
+	// +kubebuilder:validation:Required
+	Type string `json:"type"`
+
+	Datadog *DatadogIntegration `json:"datadog,omitempty"`
+
+	MicrosoftTeams *MicrosoftTeamsIntegration `json:"microsoftTeams,omitempty"`
+
+	NewRelic *NewRelicIntegration `json:"newRelic,omitempty"`
+
+	OpsGenie *OpsGenieIntegration `json:"opsGenie,omitempty"`
+
+	PagerDuty *PagerDutyIntegration `json:"pagerDuty,omitempty"`
+
+	Prometheus *PrometheusIntegration `json:"prometheus,omitempty"`
+
+	Slack *SlackIntegration `json:"slack,omitempty"`
+
+	VictorOps *VictorOpsIntegration `json:"victorOps,omitempty"`
+
+	Webhook *WebhookIntegration `json:"webhook,omitempty"`
+}
+
+type DatadogIntegration struct {
+	// APIKeySecret is the name of a secret containing the datadog api key
+	// +kubebuilder:validation:Required
+	APIKeySecret string `json:"apiKeySecret"`
+
+	// Region is the Datadog region
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+
+	// SendCollectionLatencyMetrics flags whether or not to send collection latency metrics
+	// +kubebuilder:validation:Required
+	SendCollectionLatencyMetrics bool `json:"sendCollectionLatencyMetrics"`
+
+	// SendDatabaseMetrics flags whether or not to send database metrics
+	// +kubebuilder:validation:Required
+	SendDatabaseMetrics bool `json:"sendDatabaseMetrics"`
+}
+
+type MicrosoftTeamsIntegration struct {
+	// URLSecret is the name of a secret containing the microsoft teams secret URL
+	// +kubebuilder:validation:Required
+	URLSecret string `json:"apiKeySecret"`
+}
+
+type NewRelicIntegration struct {
+	// CredentialsSecret is the name of a secret containing new relic's credentials:
+	// account id, license key, read and write tokens
+	// +kubebuilder:validation:Required
+	CredentialsSecret string `json:"credentialsSecret"`
```

Comment:
None of the new relic fields are treated as secrets by the API(are not redacted). However, as this seems to be an issue on API side, I'd say only account id is not sensitive.